starter_code/src/elevator.js

```diff
@@ -0,0 +1,102 @@
+class Elevator {
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = []; // a list of pending requests. Floors where the elevator must stop.
+    this.passengers = []; // people currently in the elevator
+    this.waitingList = []; // people waiting for the elevator
+    // this.destinationFloor = 0;
+  }
+
+  // UP & DOWN
+
+  floorUp() {
+    if (this.floor === this.MAXFLOOR) {
+      console.log("Attention la tête");
+    } else {
+      this.floor++;
+    }
+  }
+
+  floorDown() {
+    if (this.floor === 0) {
+      console.log("Do you want ants? cause this is how you get ants");
+    } else {
+      this.floor--;
+    }
+  }
+
+  // START STOP
+
+  start() {
+    this.interval = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+
+  stop() {
+    clearInterval(this.interval);
+  }
+
+  // Update the position
+  update() {
+    this.log();
+    this._passengersEnter();
+    this._passengersLeave();
+
+    // tried to set an if statement when as soon as the elevator is empty,
+    //it goes back to the bottom (floor 0) to take new people,
+    // sadly nobody want to come :( so it is stuck in some kind of infinite loop (will try to fix it)
```

Comment:
Everyone is at lunch :(
But seriously, what you can try to play with is some elevator requests that are delayed. For example, in your `index.js` try to invoke `e.call` as a callback for a setTimeout delayed a few seconds,. That way, you can check how your elevator reacts to calls that happen while it's already moving around.